### PR TITLE
Adding immutableX example

### PR DESCRIPTION
**Pull request by Saviour1001**

### CODE REVIEW

File: web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    </button>
+  );
+
+  return (
+    <div className="container">
+      <h1 className="title">
+        <a target="_blank" href="http://web3auth.io/" rel="noreferrer">
+          Web3Auth{" "}
+        </a>
+        & ReactJS Example
```
Comment (shahbaz17) [inline web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx:172]: Please update the name of the app here.

File: web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  }, []);
+
+  const registerAccounts = async () => {
+    if (!provider) {
+      uiConsole("provider not initialized yet");
+      return;
+    }
+    const rpc = new RPC(provider);
+    const response = await rpc.registerAccounts();
+    uiConsole(response);
```
Comment (shahbaz17) [inline web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx:50]: Response is empty here.

**Review by shahbaz17 — CHANGES_REQUESTED** [review]

File: web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx (hunk 2 above)
Comment (Saviour1001) [reply web-modal-sdk/immutableX/react-immutable-web3auth-example/src/App.tsx:50]: Changed the response which shows up

File: web-modal-sdk/immutableX/react-immutableX-web3auth-example/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Web3Auth x React Demo Application
```
Comment (shahbaz17) [inline web-modal-sdk/immutableX/react-immutableX-web3auth-example/README.md:1]: This still needs to change @Saviour1001

File: web-modal-sdk/immutableX/react-immutableX-web3auth-example/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    <button onClick={login} className="card">
+      Login
+    </button>
+  );
+
+  return (
+    <div className="container">
+      <h1 className="title">
+        <a target="_blank" href="http://web3auth.io/" rel="noreferrer">
+          Web3Auth{" "}
```
Comment (shahbaz17) [inline web-modal-sdk/immutableX/react-immutableX-web3auth-example/src/App.tsx:170]: Here too, please add ImmutableX.

File: web-modal-sdk/immutableX/react-immutableX-web3auth-example/src/immutableRPC.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+      const starkKey = await this.getPrivateKey();
+      const starkSigner = createStarkSigner(starkKey);
+
+      console.log("eth address", address);
+      console.log("starkAddress", starkSigner);
+
+      const walletConnection = { ethSigner, starkSigner };
+      const response = await this.client.registerOffchain(walletConnection);
+      if (response.tx_hash) {
```
Comment (shahbaz17) [inline web-modal-sdk/immutableX/react-immutableX-web3auth-example/src/immutableRPC.ts:69]: What could be the `else` statement here? If this fails?

Should we address that?

**Review by shahbaz17 — CHANGES_REQUESTED** [review]

**Review by shahbaz17 — APPROVED** [review]
LGTM

**Pull request merged**